evm/chains/sharding/mainchain_handler/vmc_utils.py

```diff
@@ -0,0 +1,212 @@
+)
+
+def mk_validation_code(address):
+    '''
+    validation_code = """
+~calldatacopy(0, 0, 128)
+~call(3000, 1, 0, 0, 128, 0, 32)
+return(~mload(0) == {})
+    """.format(utils.checksum_encode(address))
+    return serpent.compile(validation_code)
+    '''
+    # The precompiled bytecode of the validation code which
+    # verifies EC signatures
+    validation_code_bytecode = b"a\x009\x80a\x00\x0e`\x009a\x00GV`\x80`\x00`\x007` "
+    validation_code_bytecode += b"`\x00`\x80`\x00`\x00`\x01a\x0b\xb8\xf1Ps"
+    validation_code_bytecode += address
+    validation_code_bytecode += b"`\x00Q\x14` R` ` \xf3[`\x00\xf3"
+    return validation_code_bytecode
+
+def sign(message, privkey):
+    '''@privkey: Key type
+    '''
+    keys = KeyAPI()
+    signature = keys.ecdsa_sign(message, privkey)
+    v, r, s = signature.vrs
+    v += 27
+    signature_bytes = b''.join([item.to_bytes(32, 'big') for item in (v, r, s)])
+    return signature_bytes
+
+def get_valmgr_abi():
+    global _valmgr_abi, _valmgr_code
+    if not _valmgr_abi:
+        _valmgr_abi = compiler.mk_full_signature(get_valmgr_code())
+    return _valmgr_abi
+
+def get_valmgr_code():
+    global _valmgr_code
+    if not _valmgr_code:
+        mydir = os.path.dirname(__file__)
+        valmgr_path = os.path.join(mydir, '../contracts/validator_manager.v.py')
+        _valmgr_code = open(valmgr_path).read()
+    return _valmgr_code
+
+def get_valmgr_bytecode():
+    global _valmgr_bytecode
+    if not _valmgr_bytecode:
+        _valmgr_bytecode = compiler.compile(get_valmgr_code())
+    return _valmgr_bytecode
+
+def get_valmgr_addr():
+    global _valmgr_addr
+    if not _valmgr_addr:
+        create_valmgr_tx()
+    return _valmgr_addr
+
+def get_valmgr_sender_addr():
+    global _valmgr_sender_addr
+    if not _valmgr_sender_addr:
+        create_valmgr_tx()
+    return _valmgr_sender_addr
+
+def get_valmgr_tx():
+    global _valmgr_tx
+    if not _valmgr_tx:
+        create_valmgr_tx()
+    return _valmgr_tx
+
+def create_valmgr_tx(gasprice=GASPRICE):
+    global _valmgr_sender_addr, _valmgr_addr, _valmgr_tx
+    bytecode = get_valmgr_bytecode()
+    v = 27
+    r = 1000000000000000000000000000000000000000000000000000000000000000000000000000
+    s = 1000000000000000000000000000000000000000000000000000000000000000000000000000
+    tx = SpuriousDragonTransaction(0, gasprice, 3000000, b'', 0, bytecode, v, r, s)
+    valmgr_sender_addr = tx.sender
+    valmgr_addr = eth_utils.to_checksum_address(
+        generate_contract_address(eth_utils.to_canonical_address(valmgr_sender_addr), 0)
+    )
+    _valmgr_sender_addr = eth_utils.to_checksum_address(valmgr_sender_addr)
+    _valmgr_addr = valmgr_addr
+    _valmgr_tx = tx
+
+def mk_initiating_contracts(sender_privkey, sender_starting_nonce, gasprice=GASPRICE):
+    """Make transactions of createing initial contracts
+    Including rlp_decoder, sighasher and validator_manager
+    """
+    o = []
```

Comment:
In general I'd like to avoid single letter variable names.

This function however can be adjusted to use `eth_utils.to_list`.  All of the places where you `o.append` or `o += [...]` can be turned into `yield` statements and the resulting return from this function will be all of those values that were yielded in a list.

```python
for tx in (....):
    yield SpuriousDragonUnsignedTransaction(...)
yield viper_rlp_decoder_tx
yield sighasher_tx
...
```

Reply:
You're right. I'm sorry for that.
I will modify it to `yield` values and use `@to_tuple` to decorate the function.